erpnext/manufacturing/report/cost_of_poor_quality_report/cost_of_poor_quality_report.py

```diff
@@ -18,31 +17,45 @@ def get_data(report_filters):
 	data = []
 	operations = frappe.get_all("Operation", filters = {"is_corrective_operation": 1})
 	if operations:
-		operations = [d.name for d in operations]
-		fields = ["production_item as item_code", "item_name", "work_order", "operation",
-			"workstation", "total_time_in_mins", "name", "hour_rate", "serial_no", "batch_no"]
 
-		filters = get_filters(report_filters, operations)
-
-		job_cards = frappe.get_all("Job Card", fields = fields,
-			filters = filters)
-
-		for row in job_cards:
-			row.operating_cost = flt(row.hour_rate) * (flt(row.total_time_in_mins) / 60.0)
-			data.append(row)
+		if report_filters.get('operation'):
+			operations = [report_filters.get('operation')]
+		else:
+			operations = [d.name for d in operations]
 
+		job_card_time_log = frappe.qb.DocType("Job Card Time Log")
+		job_card = frappe.qb.DocType("Job Card")
+		operatingcost = ((job_card.hour_rate) * (job_card.total_time_in_mins) / 60.0).as_('operating_cost')
+		itemcode = (job_card.production_item).as_('item_code')
+		query = (
+					frappe.qb.from_(job_card)
+					.select(job_card.name, job_card.work_order, itemcode, job_card.item_name, job_card.operation,
+					job_card.serial_no, job_card.batch_no, job_card.workstation, job_card.total_time_in_mins, job_card.hour_rate,
+					operatingcost)
+					.where(job_card.docstatus==1)
+					.where(job_card.is_corrective_job_card==1)
```

Comment:
- This condition should also be added in the subquery for faster filtering. 
- Sub-query should only be required when from/to time is specified.